From 2edc3cf8799eecae9cc476611f1259923314f6f8 Mon Sep 17 00:00:00 2001
From: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>
Date: Wed, 21 Dec 2016 16:07:27 +0200
Subject: [PATCH] configure.ac: add option to disable mtd_probe

FTL is not necessarily present on all kernels so it's useful to
have an option to disable this.

Signed-off-by: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>

Upstream-Status: Backport [commit 2edc3cf8799 on upstream master]

---
 configure.ac      | 8 ++++++++
 rules/Makefile.am | 6 +++++-
 src/Makefile.am   | 6 +++++-
 3 files changed, 18 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index ff90fe488..52865a815 100644
--- a/configure.ac
+++ b/configure.ac
@@ -284,6 +284,14 @@ fi
 AM_CONDITIONAL([ENABLE_RULE_GENERATOR], [test "x$enable_rule_generator" = xyes])
 
 # ------------------------------------------------------------------------------
+# mtd_probe - autoloads FTL module for mtd devices
+# ------------------------------------------------------------------------------
+AC_ARG_ENABLE([mtd_probe],
+	AS_HELP_STRING([--disable-mtd_probe], [disable MTD support]),
+	[], [enable_mtd_probe=yes])
+AM_CONDITIONAL([ENABLE_MTD_PROBE], [test "x$enable_mtd_probe" = xyes])
+
+# ------------------------------------------------------------------------------
 
 AC_CONFIG_FILES([Makefile
                  hwdb/Makefile
diff --git a/rules/Makefile.am b/rules/Makefile.am
index a403710ed..e0b17d2ca 100644
--- a/rules/Makefile.am
+++ b/rules/Makefile.am
@@ -16,7 +16,6 @@ dist_udevrules_DATA = \
 	70-mouse.rules \
 	70-touchpad.rules \
 	75-net-description.rules \
-	75-probe_mtd.rules \
 	78-sound-card.rules
 
 if !ENABLE_RULE_GENERATOR
@@ -34,5 +33,10 @@ dist_udevrules_DATA += \
 	80-drivers.rules
 endif
 
+if ENABLE_MTD_PROBE
+dist_udevrules_DATA += \
+	75-probe_mtd.rules
+endif
+
 install-data-local:
 	$(MKDIR_P) $(DESTDIR)$(udevconfdir)/rules.d
diff --git a/src/Makefile.am b/src/Makefile.am
index f0ddd0125..fc052a83e 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -10,6 +10,10 @@ SUBDIRS += \
 	ata_id \
 	cdrom_id \
 	collect \
-	mtd_probe \
 	scsi_id \
 	v4l_id
+
+if ENABLE_MTD_PROBE
+SUBDIRS += \
+	mtd_probe
+endif
-- 
2.11.0

